CIScripts/Test/NewContainer.Tests.ps1

```diff
@@ -0,0 +1,127 @@
+            Set-Content -Path "docker.ps1" -Value $DockerThatAlwaysSucceeds
+        }
+
+        $NewContainerID = New-Container `
+            -Session $Session `
+            -NetworkName "BestNetwork" `
+            -Name "jolly-lumberjack" | Should -Not -Throw
+        $NewContainerID | Should -Be $ContainerID
+    }
+
+    It "Throws exception when container creation fails" {
+        Invoke-Command -Session $Session {
+            $DockerThatAlwaysFails = @"
+            Write-Error "It's Docker here: This is very very bad!"
+            exit 1
+"@
+            Set-Content -Path "docker.ps1" -Value $DockerThatAlwaysFails
+        }
+
+        New-Container `
+            -Session $Session `
+            -NetworkName "NetworkOfFriends" `
+            -Name "jolly-lumberjack" | Should -Throw
+    }
+
+    It "Reports container id when container creation succeeds in second attempt after failing because of known issue" {
+        Invoke-Command -Session $Session {
+            $DockerThatSucceedsInSecondAttempt = @'
+            if ($args[0] -eq "run") {{
+                $Hope = "HopeForSuccess"
+                if (Test-Path $Hope) {{
+                    Write-Output "{0}"
+                    Remove-Item $Hope
+                    exit 0
+                }} else {{
+                    Set-Content -Path $Hope -Value "New hope"
+                    Write-Output "{0}"
+                    Write-Error "It's Docker here: CreateContainer: failure in a Windows system call. Try again. Good luck!"
+                    exit 1
+                }}
+            }} else {{
+                Write-Output "{0}"
+                exit 0
+            }}
+'@ -f $Using:ContainerID
+            Set-Content -Path "docker.ps1" -Value $DockerThatSucceedsInSecondAttempt
+        }
+
+        $NewContainerID = New-Container `
+            -Session $Session `
+            -NetworkName "SoftwareDefinedNetwork" `
+            -Name "jolly-lumberjack" | Should -Not -Throw
+        $NewContainerID | Should -Be $ContainerID
+    }
+
+    It "Throws exception when container creation fails in first attempt and reports unknown issue" {
+        Invoke-Command -Session $Session {
+            $HopeStorage = "HopeForLuckyTry"
+            Remove-Item $HopeStorage -ErrorAction Ignore
+            $DockerThatSucceedsInSecondAttempt = @'
+            if ($args[0] -eq "run") {{
+                $Hope = "{1}"
+                if (Test-Path $Hope) {{
+                    Write-Output "{0}"
+                    Remove-Item $Hope
+                    exit 0
+                }} else {{
+                    Set-Content -Path $Hope -Value "There's actually no hope."
+                    Write-Output "{0}"
+                    Write-Error "It's Docker here: Error code #190583. Shall you try again?"
+                    exit 1
+                }}
+            }} else {{
+                Write-Output "{0}"
+                exit 0
+            }}
+'@ -f $Using:ContainerID,$HopeStorage
```

Comment:
Decide on one mechanism for supplying the path - either hardcoded or passed

Reply:
It's hardcoded when there is only one piece of code that is using a path - as simple as possible.
It's passed as a variable when it's used in multiple places.